library/src/main/scala/scala/scalajs/js/annotation/LinkTime.scala

```diff
@@ -0,0 +1,25 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.scalajs.js.annotation
+
+/** Specifies that the entity can be replaced by a value known at link-time.
+ *
+ *  The annotated entity can be used in the condition of `LinkingInfo.linkTimeIf`.
+ *  When linking, the annotated entity will be replaced by a value determined
+ *  at link-time, as specified in `LinkTimeProperties.scala`.
+ *
+ *  @see [[LinkingInfo.linkTimeIf]]
+ *  @see [[LinkTimeProperties]]
+ */
+private[scalajs] class LinkTime extends scala.annotation.StaticAnnotation
```

Comment:
Other comment here: IMO this annotation should take a string argument which will be used as its IR encoding. We shouldn't rely on the fully-qualified name of a symbol, as it has several constraints:
* we cannot define aliases
* we necessarily embed the `scala.*` namespace in the names of core properties, whereas the IR is supposed to be Scala-agnostic.

Reply:
Thanks, that makes sense :+1:
Now `@linkTimeProperty` annotation requires an argument that represent a key of the link-time values dictionary (such as `core/productionMode` and `core/esVersion`).